### PR TITLE
FIX: realm encryption mismatch after backup restore (closes #3332)

### LINKED ISSUE #3332: Save to disk exception - realm file decryption failed
I had bluewallet on an iPhone 7. I migrated to iPhone 11. When I launch bluewallet I get this:
image

The iPhone 11 has some MDM restrictons.

What do?

### PR DESCRIPTION
**Pull request by Overtorment**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/kfcv2k8hd1jt140ajpmzp0phjr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7779)

File: class/app-storage.js [hunk 1]
```diff
@@ -580,6 +580,10 @@ export class AppStorage {
     } catch (error) {
       console.error('save to disk exception:', error.message);
       alert('save to disk exception: ' + error.message);
+      if (error.message.includes('Realm file decryption failed')) {
+        console.warn('purging realm key-value database file');
+        this.purgeRealmKeyValueFile();
```
Comment (limpbrains) [inline class/app-storage.js:585]: Is it ok to call async this.purgeRealmKeyValueFile() method without await ?

**Review by limpbrains — CHANGES_REQUESTED** [review]

File: class/app-storage.js (hunk 1 above)
Comment (Overtorment) [reply class/app-storage.js:585]: good point. `Realm.deleteFile` is not async.
fixed

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/040ne4v14qfjnpxvy043fbx1m4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7780)

**Review by limpbrains — APPROVED** [review]

**Review by marcosrdz — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**